sbol_utilities/component.py

```diff
@@ -519,3 +519,141 @@ def ed_protein(definition: str, **kwargs) -> sbol3.ExternallyDefined:
     :return: A Component object.
     """
     return sbol3.ExternallyDefined([sbol3.SBO_PROTEIN], definition, **kwargs)
+
+def ed_restriction_enzyme(definition: str, name:str, **kwargs) -> sbol3.ExternallyDefined:
+    """Creates an ExternallyDefined Restriction Enzyme Component.
+
+    :param definition: The URI that links to a canonical definition external to SBOL, recommended UniProt.
+    :param name: Name of the SBOL ExternallyDefined, used by PyDNA. Follow standard restriction enzyme nomenclature, i.e. 'BsaI'
+    :param kwargs: Keyword arguments of any other ExternallyDefined attribute.
+    :return: A Component object.
+    """
+    return sbol3.ExternallyDefined([sbol3.SBO_PROTEIN], definition, name=name **kwargs)
```

Comment:
Add a test in Bio.Restriction to make sure the enzyme can be looked up there.
If it can't be, then it's not a standard restriction enzyme, and should raise a ValueError.

Reply:
Implemented using validators.url
```
    definition=f'http://rebase.neb.com/rebase/enz/{name}.html'
    if not validators.url(definition): 
        raise ValueError
```

Reply:
I tested validators and url with other names not only the correct format is taken as valid, I need to find another way to validate. Changed to
```
from Bio.Restriction import name
definition=f'http://rebase.neb.com/rebase/enz/{name}.html'
```